botocore/utils.py

```diff
@@ -925,12 +925,33 @@ def set_request_url(self, params, context, **kwargs):
         if endpoint is not None:
             params['url'] = _get_new_endpoint(params['url'], endpoint)
 
-    def redirect_from_cache(self, params, context, **kwargs):
+    def modify_request_before_send(self, params, context, model, **kwargs):
+        bucket = params.get('Bucket')
+        self.redirect_from_cache(bucket, context)
+        if model.name == 'CreateBucket':
+            self.redirect_create_bucket(bucket, params, context)
+
+    def redirect_create_bucket(self, bucket, params, context):
+        configuration = params.get('CreateBucketConfiguration', {})
+        location = configuration.get('LocationConstraint', None)
+
+        if location is not None:
```

Comment:
That I'm actually not sure about. I'm fairly certain it's only good for us-east-1.